presentation/src/main/java/com/keeply/presentation/ui/my/setting/alert/AlertSettingScreen.kt

```diff
@@ -1,29 +1,123 @@
 import com.keeply.presentation.ui.my.setting.component.SettingToggleItem
+import org.orbitmvi.orbit.compose.collectAsState
+import org.orbitmvi.orbit.compose.collectSideEffect
 
 @Composable
 fun SettingAlertRoute(
     viewModel: SettingAlertViewModel = hiltViewModel(),
+    onBackClick: () -> Unit = {}
 ) {
-    SettingAlertScreen()
+    val state by viewModel.collectAsState()
+    val context = LocalContext.current
+    
+    // 권한 요청 런처
+    val notificationPermissionLauncher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.RequestPermission()
+    ) { isGranted ->
+        viewModel.updateNotificationPermission(isGranted)
+    }
+    
+    // 권한 상태 확인 및 업데이트
+    LaunchedEffect(Unit) {
+        val hasPermission = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            NotificationManagerCompat.from(context).areNotificationsEnabled()
+        } else {
+            NotificationManagerCompat.from(context).areNotificationsEnabled()
+        }
+        viewModel.updateNotificationPermission(hasPermission)
+    }
+    
+    // 화면이 다시 포커스될 때 권한 상태 재확인
+    val lifecycleOwner = LocalLifecycleOwner.current
+    DisposableEffect(lifecycleOwner) {
+        val observer = LifecycleEventObserver { _, event ->
+            if (event == Lifecycle.Event.ON_RESUME) {
+                val hasPermission = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                    NotificationManagerCompat.from(context).areNotificationsEnabled()
+                } else {
+                    NotificationManagerCompat.from(context).areNotificationsEnabled()
+                }
+                viewModel.updateNotificationPermission(hasPermission)
+            }
+        }
+        lifecycleOwner.lifecycle.addObserver(observer)
+        
+        onDispose {
+            lifecycleOwner.lifecycle.removeObserver(observer)
+        }
+    }
+    
+    viewModel.collectSideEffect { sideEffect ->
+        when (sideEffect) {
+            is SettingAlertSideEffect.ShowError -> {
+                // TODO: Show error message (예: SnackBar, Toast 등)
+            }
+            is SettingAlertSideEffect.RequestNotificationPermission -> {
+                // 권한이 없을 때: 권한 요청 모달 표시
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                    notificationPermissionLauncher.launch(Manifest.permission.POST_NOTIFICATIONS)
+                } else {
+                    // Android 12 이하에서는 알림이 기본적으로 허용됨
+                    viewModel.updateNotificationPermission(true)
+                }
+            }
+            is SettingAlertSideEffect.OpenNotificationSettings -> {
+                // 권한이 있을 때: 설정 화면으로 이동하여 사용자가 직접 끌 수 있도록
+                val intent = Intent().apply {
+                    action = Settings.ACTION_APPLICATION_DETAILS_SETTINGS
+                    data = Uri.fromParts("package", context.packageName, null)
+                }
+                context.startActivity(intent)
```

Comment:
p2: PermissionUtil.kt 의 함수를 이용해도 좋을 것 같습니당
```suggestion
                openAppSettings(context)
```

Reply:
오 좋아요~